### DISCUSSION TITLE
Android Emulator Webview PageSource is 'stuck'?

### DISCUSSION THREAD

**Discussion by snoopgr2** [appium/appium#18981] [Q&A]
While working with an Android Simulator (Pixel 6 API 33 and some older versions) I find that my Test fail constantly.  With research I found that the Pagesource of my WebHybrid app get's 'stuck'.  By this I mean that if I launch a 3rd party webpage (like google.com) inside of my webview, the emulator seems to keep the new pagesource in memory.  Now, this is only for the Appium interface with UIAutomator2.  By this I mean that the application continues to work with manual clicks and scrolling. But Appium can no longer access the correct pagesource after I return to our pages.

Has anyone seen anything like this and if so, how I do get around this.  Android is completely unreliable due to this as our login process is a separate web page and creates this situation in between almost every test I do.

I should point out that the only way I've found to clear the pagesource from the emulator is to stop the emulator and do a cold boot and hope that after the login process I'm not stuck again.  

This really seems to be a problem in UIAutomator2 as Appium is connecting and able to read the pagesource with no issues but the pagesource is incorrect.

**Comment by jlipps** [comment]
Please create an issue by filling out the template with all the requested information
